airtime_mvc/application/forms/LiveStreamingPreferences.php

```diff
@@ -125,12 +138,12 @@ public function updateVariables()
                 array ('ViewScript',
                     array (
                       'viewScript'                  => 'form/preferences_livestream.phtml',
-                      'master_source_host'          => isset($masterSourceParams["host"])?$masterSourceParams["host"]:"",
-                      'master_source_port'          => isset($masterSourceParams["port"])?$masterSourceParams["port"]:"",
-                      'master_source_mount'         => isset($masterSourceParams["path"])?$masterSourceParams["path"]:"",
-                      'show_source_host'            => isset($showSourceParams["host"])?$showSourceParams["host"]:"",
-                      'show_source_port'            => isset($showSourceParams["port"])?$showSourceParams["port"]:"",
-                      'show_source_mount'           => isset($showSourceParams["path"])?$showSourceParams["path"]:"",
+                      'master_source_host'          => isset($masterSourceHost)?$masterSourceParams["host"]:"",
+                      'master_source_port'          => isset($masterSourcePort)?$masterSourceParams["port"]:"",
+                      'master_source_mount'         => isset($masterSourceMount)?$masterSourceParams["path"]:"",
+                      'show_source_host'            => isset($showSourceHost)?$showSourceParams["host"]:"",
+                      'show_source_port'            => isset($showSourcePort)?$showSourceParams["port"]:"",
+                      'show_source_mount'           => isset($showSourceMount)?$showSourceParams["path"]:"",
                       'isDemo'                      => $isDemo,
                     )
```

Comment:
Ok that makes sense. I wasn't familiar with how the function parse_url worked. Perhaps that is a more elegant solution than putting everything in as a separate key, but I think that the python liquidsoap reconfigure function depends upon them being a separate key for its checks. So without refactoring that.